opslevel/resource_opslevel_check_tag_defined.go

```diff
@@ -1,83 +1,220 @@
-// 	input.TagKey = d.Get("tag_key").(string)
-// 	input.TagPredicate = expandPredicate(d, "tag_predicate")
-
-// 	resource, err := client.CreateCheckTagDefined(*input)
-// 	if err != nil {
-// 		return err
-// 	}
-// 	d.SetId(string(resource.Id))
-
-// 	return resourceCheckTagDefinedRead(d, client)
-// }
-
-// func resourceCheckTagDefinedRead(d *schema.ResourceData, client *opslevel.Client) error {
-// 	id := d.Id()
-
-// 	resource, err := client.GetCheck(opslevel.ID(id))
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	if err := setCheckData(d, resource); err != nil {
-// 		return err
-// 	}
-// 	if err := d.Set("tag_key", resource.TagKey); err != nil {
-// 		return err
-// 	}
-// 	if err := d.Set("tag_predicate", flattenPredicate(resource.TagPredicate)); err != nil {
-// 		return err
-// 	}
-
-// 	return nil
-// }
-
-// func resourceCheckTagDefinedUpdate(d *schema.ResourceData, client *opslevel.Client) error {
-// 	checkUpdateInput := getCheckUpdateInputFrom(d)
-// 	input := opslevel.NewCheckUpdateInputTypeOf[opslevel.CheckTagDefinedUpdateInput](checkUpdateInput)
-
-// 	if d.HasChange("tag_key") {
-// 		input.TagKey = opslevel.RefOf(d.Get("tag_key").(string))
-// 	}
-// 	if d.HasChange("tag_predicate") {
-// 		input.TagPredicate = expandPredicateUpdate(d, "tag_predicate")
-// 	}
-
-// 	_, err := client.UpdateCheckTagDefined(*input)
-// 	if err != nil {
-// 		return err
-// 	}
-// 	d.Set("last_updated", timeLastUpdated())
-// 	return resourceCheckTagDefinedRead(d, client)
-// }
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+	"github.com/relvacode/iso8601"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &CheckTagDefinedResource{}
+	_ resource.ResourceWithImportState = &CheckTagDefinedResource{}
+)
+
+func NewCheckTagDefinedResource() resource.Resource {
+	return &CheckTagDefinedResource{}
+}
+
+// CheckTagDefinedResource defines the resource implementation.
+type CheckTagDefinedResource struct {
+	CommonResourceClient
+}
+
+type CheckTagDefinedResourceModel struct {
+	Category    types.String `tfsdk:"category"`
+	Description types.String `tfsdk:"description"`
+	Enabled     types.Bool   `tfsdk:"enabled"`
+	EnableOn    types.String `tfsdk:"enable_on"`
+	Filter      types.String `tfsdk:"filter"`
+	Id          types.String `tfsdk:"id"`
+	Level       types.String `tfsdk:"level"`
+	Name        types.String `tfsdk:"name"`
+	Notes       types.String `tfsdk:"notes"`
+	Owner       types.String `tfsdk:"owner"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+
+	TagKey       types.String   `tfsdk:"tag_key"`
+	TagPredicate PredicateModel `tfsdk:"tag_predicate"`
```

Comment:
Why do we use a concrete type here, but here we used a pointer? https://github.com/OpsLevel/terraform-provider-opslevel/pull/274/files#diff-b169d5f414a75d5eb34295e216327d463f9eddcada9b63054250c3ed9d115c80R49

Reply:
good catch